tests/animalShelter.test.ts

```diff
@@ -167,4 +178,56 @@ describe('Animal Shelter Tests', () => {
     expect(animals.animalsWhoJump).toBe(2)
     expect(animals.animalsWhoRun).toBe(1)
   })
+  test('report should return an object with the correct number of animals with specific abilities', () => {
+    const animalShelter = AnimalShelter.getInstance()
+    const tortoise = new Tortoise('t1', 3, false)
+    const tortoise2 = new Tortoise('t2', 4, false)
+    const squirrel = new Squirrel('s1', 3, false, 'Maple', 44, 4)
+    const lion = new Lion('simba', 4, true)
+    animalShelter.addAnimal(tortoise)
+    animalShelter.addAnimal(tortoise2)
+    animalShelter.addAnimal(squirrel)
+    animalShelter.addAnimal(lion)
+    const animals = animalShelter.report()
+    expect(animals.anaimalsWhoClimbTrees).toBe(4)
+    expect(animals.animalsWhoJump).toBe(2)
+    expect(animals.animalsWhoRun).toBe(1)
+  })
+
+  test('should return an estimate of future space needed if females always give birth', () => {
```

Comment:
we have an explicit requirement to report for 5 years in the future, yet we don't even have a single test on this functionality